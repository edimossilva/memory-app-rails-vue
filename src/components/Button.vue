<template>
  <button @click="onClick">{{ label }}</button>
</template>

<script>
export default {
  name: "Button",
  props: {
    label: {
      type: String,
      required: true,
    },

    onClick: {
      type: Function,
      required: false,
      default: () => {},
    },
  },
};
</script>
