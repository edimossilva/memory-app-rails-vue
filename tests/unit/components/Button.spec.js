import { shallowMount } from "@vue/test-utils";
import Button from "@/components/Button.vue";
import Vue from "vue";
import VueRouter from "vue-router";

Vue.use(VueRouter);
describe("Button.vue", () => {
  it("Matchs Snapshot", () => {
    const wrapper = shallowMount(Button, {
      propsData: { label: "mock label" },
    });
    expect(wrapper).toMatchSnapshot();
  });
});
