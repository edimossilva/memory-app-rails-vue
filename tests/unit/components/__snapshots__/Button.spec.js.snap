// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button.vue Matchs Snapshot 1`] = `<button>mock label</button>`;
